### PR TITLE
Burrowed larva and larva now count towards xenomorph evolution tier and queen calculation

### PR DESCRIPTION
**Pull request by ma44**
## About The Pull Request

Before burrowed larva and regular larva never counted towards the 8 xeno threshold for queen and for evolving into a tier 2 or 3 xeno; now it does

## Why It's Good For The Game
Prevents xenos getting screwed over from being able to evolve if theres

- No one willing to play burrowed larva
- SSD larvas that don't evolve

## Changelog
:cl:
balance: Burrowed and regular larva will now count towards evolution equations, meaning xenomorphs can evolve as if burrowed and unburrowed larva were actual xenomorphs possessed by players
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/evolution.dm [hunk 1]
```diff
@@ -188,14 +188,18 @@
...
 
 	else
 		var/potential_queens = length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/larva]) + length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/drone])
 
-		tierones = length(hive.xenos_by_tier[XENO_TIER_ONE])
+		if(istype(hive, /datum/hive_status/normal)) //If it has burrowed larva let's include that
+			var/datum/hive_status/normal/hive2
+			tierones = length(hive2.xenos_by_tier[XENO_TIER_ONE]) + length(hive2.xenos_by_tier[XENO_TIER_ZERO]) + length(hive2.stored_larva) //Includes larva and burrowed larva
+		else
+			tierones = length(hive.xenos_by_tier[XENO_TIER_ONE]) + length(hive.xenos_by_tier[XENO_TIER_ZERO])
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/evolution.dm]:
```suggestion
			tierones = length(hive.xenos_by_tier[XENO_TIER_ONE]) + length(hive.xenos_by_tier[XENO_TIER_ZERO])
			if(istype(hive, /datum/hive_status/normal))
							var/datum/hive_status/normal/normal_hive = hive
							tierones += normal_hive.stored_larva
```

* `var/datum/hive_status/normal/hive2` is undefined, but a more descriptive name like `normal_hive` is preferable anyway.
* `stored_larva` is not a list, it's already a numeric value.

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 2]
```diff
@@ -46,6 +46,13 @@
 	for(var/t in xenos_by_tier)
 		. += length(xenos_by_tier[t])
 
+//Normal hive status includes burrowed
+/datum/hive_status/normal/get_total_xeno_number()
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm]: This proc is used elsewhere, so what you want to redefine is `can_hive_have_a_queen()`

File: code/modules/mob/living/carbon/xenomorph/evolution.dm [hunk 3]
```diff
@@ -21,7 +21,7 @@
 	set name = "Regress"
 	set desc = "Regress into a lower form."
 	set category = "Alien"
-	
+
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/evolution.dm]: avoid committing whitespace changes please

File: code/modules/mob/living/carbon/xenomorph/evolution.dm [hunk 4]
```diff
@@ -188,14 +188,17 @@
...
-		
+
 		if(mind)
 			mind.assigned_role = ROLE_XENO_QUEEN
 
 	else
 		var/potential_queens = length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/larva]) + length(hive.xenos_by_typepath[/mob/living/carbon/xenomorph/drone])
 
-		tierones = length(hive.xenos_by_tier[XENO_TIER_ONE])
+		tierones = length(hive.xenos_by_tier[XENO_TIER_ONE]) + length(hive.xenos_by_tier[XENO_TIER_ZERO])
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/evolution.dm]: this is just wacked, make a var/tierzeros

File: code/modules/mob/living/carbon/xenomorph/evolution.dm [hunk 5]
```diff
@@ -188,14 +188,17 @@
...
+		if(istype(hive, /datum/hive_status/normal))
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/evolution.dm]: dont do this make a datum/hive_status/ proc instead

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 6]
```diff
@@ -69,6 +69,8 @@
 /datum/hive_status/proc/can_hive_have_a_queen()
 	return get_total_xeno_number() < xenos_per_queen
 
+/datum/hive_status/normal/can_hive_have_a_queen()
+	return get_total_xeno_number() + stored_larva < xenos_per_queen
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm]: just override get_total_xeno_number()

Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/hive_datum.dm]: Check the previous change requests.
`get_total_xeno_number()` is used on a different proc.

Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm]:
```suggestion
	return ((get_total_xeno_number() + stored_larva) < xenos_per_queen)
```

**Pull request merged**